### PR TITLE
Docs: Create documentation for enterprise caching HTTP API

### PR DESCRIPTION
**Pull request by mmandrus**
**What is this feature?**

Creates documentation for the query and resource caching HTTP API

**Why do we need this feature?**

Customer request.

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana-enterprise/issues/4957

**Special notes**

Please give an extra look to the metadata section at the top, I may have overdone it with aliases and key words

### CODE REVIEW

**Comment by osg-grafana** [comment]
@Eve832, as triager this week, I have put this into the team project and into In review.

File: docs/sources/developers/http_api/query_and_resource_caching.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  - caching
+  - query caching
+  - resource caching
+  - data source
+title: Query and Resource Caching HTTP API
+---
+
+# Query and resource caching API
+
+> If you are running Grafana Enterprise, for some endpoints you'll need to have specific permissions. Refer to [Role-based access control permissions]({{< relref "/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes" >}}) for more information.
```
Comment (Eve832) [inline docs/sources/developers/http_api/query_and_resource_caching.md:22]:
```suggestion
>**Note:** If you are running Grafana Enterprise, for some endpoints you'll need to have specific permissions. Refer to [Role-based access control permissions]({{< relref "/docs/grafana/latest/administration/roles-and-permissions/access-control/custom-role-actions-scopes" >}}) for more information.
```

**Review by Eve832 — APPROVED** [review]
Thank you for adding this! 
Just one minor nit and two questions: 
- does this apply to Cloud users or only on-prem?
- Which version of Grafana does this ship with?

**Comment by mmandrus** [comment]
> Thank you for adding this! Just one minor nit and two questions:
> 
> * does this apply to Cloud users or only on-prem?
> * Which version of Grafana does this ship with?

@Eve832 
1. this is for both on-prem and cloud. note that I copied from another HTTP API page, so the disclaimers/notes are consistent with what the rest of our docs say
2. This feature has existed for a long time, it was just never documented. These docs apply to all currently supported versions of Grafana.

**Pull request merged**